src/test/ui/impl-trait/multiple-lifetimes/ordinary-bounds-unrelated.nll.stderr

```diff
@@ -4,7 +4,11 @@ error[E0700]: hidden type for `impl Trait` captures lifetime that does not appea
 LL | fn upper_bounds<'a, 'b, 'c, 'd, 'e>(a: Ordinary<'a>, b: Ordinary<'b>) -> impl Trait<'d, 'e>
    |                                                                          ^^^^^^^^^^^^^^^^^^
    |
-   = note: hidden type `Ordinary<'_>` captures lifetime '_#9r
+note: hidden type `Ordinary<'b>` captures the lifetime `'b` as defined on the function body at 16:21
```

Comment:
(This is pre-existing) 

This message mentions `Ordinary<'b>` here as a "hidden type", shouldn't this be "hidden type `impl Trait`" instead, or am I misunderstanding this diagnostic ?

Reply:
You are completely right.

We should completely rewrite the diagnostics around TAIT in general... they are all not too helpful. I'll open an issue to track the various ways in which it is not great